### DISCUSSION TITLE
Nightscout Follower - 1 min intervall issue

### DISCUSSION THREAD

**Discussion by PythonFZ** [NightscoutFoundation/xDrip#2225] [General]
I'm using https://github.com/timoschlueter/nightscout-librelink-up with a 1 min intervall to upload to nightscout.
I then use the nightscout follower option in xDrip and save meals / bolus and upload it to nightscout.
Since I changed the intervall in the librelinkup to 1 min xDrip struggles to display the values correctly.
![image](https://user-images.githubusercontent.com/46721498/179351094-4b1711ea-a7ea-4760-9e0c-324bdfc0923c.png)
It only shows the last 1/5 of the history. Is it possible to somehow adapt this? I would be fine with only having every 5 min updates in xDrip but for smoother plots I prefer the 1 min intervall in nightscout.

**Comment by tzachi-dar** [comment]
Can you please zoom in to one hour and show the display? (I'm trying to understand if xDrip is showing a point every minute or every 5 minutes)

**Comment by PythonFZ** [comment]
![image](https://user-images.githubusercontent.com/46721498/179694292-1d75f267-f090-4158-829a-35b29b3f3deb.png)

This is as far as I can zoom in. Are there any settings I could try to?

**Reply by Navid200** [reply]
Sorry, that's good.

**Reply by Navid200** [reply]
If it was showing only one point every 5 minutes, you would only see 60 / 5 = 12 points in one hour.
It is showing more.

It is showing all the 60 points, which means one every minute.  But, it has arranged them on a 2.5-minute grid.

Would you please do an experiment?  
Please enable engineering mode:  https://navid200.github.io/xDrip/docs/Engineering-Mode.html
Then, enable Settings -> xDrip Display Settings -> Graph Settings -> Lower Fuzzer

Restart the phone.
What does it show now?

**Reply by PythonFZ** [reply]
I enabled the engineering mode and it was displayed on the main screen that is is active. After enabling `Lower Fuzzer` the message was not shown anymore. After restarting `Lower Fuzzer` is still selected in the settings but I can not see any immediate change. If this only affects future values I will report on it later/tomorrow.

**Comment by Navid200** [comment]
There are two different factors at play here.
One is the points showing up on a 2.5-minute grid instead of on a 1-minute grid.
The second factor is the missing points.

The second factor is mentioned in this issue:  https://github.com/NightscoutFoundation/xDrip/issues/1518#issuecomment-731638171

Both factors are important.  My questions are meant to address the first factor hoping to get more info so that we may figure out how to deal with it.

**Comment by tzachi-dar** [comment]
IMHO, the solution that is needed here is to find where the BGs are being added to xDrip and make sure not to add a new BG if there is another one 5 minutes from there.
Something like: https://github.com/NightscoutFoundation/xDrip/blob/9fcaafeb729c0eb0e0988fbd96322d292649efd4/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java#L1269

**Comment by jakobheyman** [comment]
I believe the 2.5 min grid issue can be resolved with this commit:
https://github.com/NightscoutFoundation/xDrip/pull/1401/commits/d46f355830aa89f5f7b2f66b163df32cc2b9cf28

The missing points issue can perhaps be fixed by changing NUM_VALUES (not dividing by 5) here:
https://github.com/NightscoutFoundation/xDrip/blob/9fcaafeb729c0eb0e0988fbd96322d292649efd4/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java#L114

**Reply by tzachi-dar** [reply]
This will solve one immediate issue, but other issues will exist. It is better to make sure that bad data will not get in.
This has influence on alerts for examples.

**Reply by Navid200** [reply]
@jakobheyman Can you open a PR with your proposed fix?  Can you explain in it the fix, how you have tested it, and how we can be sure that it will not affect other devices like Dexcom transmitters etc?

**Reply by jakobheyman** [reply]
@Navid200 I can open a PR if you wish. However, my comment was mostly a best guess based on the code and I have not tested it properly.

**Comment by Navid200** [comment]
@PythonFZ Would you please try the xDrip version that has just been released (https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.07.21) and let us know if you see any difference?

**Reply by PythonFZ** [reply]
It seems to work now in the main screen. Very nice and thanks to all who participated! Unfortunately, looking at the History tab the issue seems to remain.

**Reply by Navid200** [reply]
I don't think it is going to go back in time and reload all the missing data.
Would you please report tomorrow if then looking back at today, you see a problem?  Then, we have a problem with history.

**Reply by PythonFZ** [reply]
The history still only shows part of the data, starting at about 11 am to midnight.

**Reply by Navid200** [reply]
Let me look into that.

**Reply by Navid200** [reply]
Can I see a screenshot please?

**Reply by PythonFZ** [reply]
Sure, here are 1 day and 2 days of history. Both after using the newest version.
![image](https://user-images.githubusercontent.com/46721498/180781221-4f70d7c2-8a6e-4688-873e-0dff8d3123d7.png)
![image](https://user-images.githubusercontent.com/46721498/180781254-652bd11f-09bb-413d-a41a-26ce6da9e0e0.png)

**Reply by Navid200** [reply]
Thanks.  Would you please zoom in as much as you can so that I can see how often the readings appear?

**Reply by PythonFZ** [reply]
Seems to be about 24 x-ticks per hour with sometimes multiple values.

![image](https://user-images.githubusercontent.com/46721498/180781904-cc267f53-582e-417d-9f9f-357e2c903357.png)

**Reply by Navid200** [reply]
Thanks

**Reply by jakobheyman** [reply]
My guess would be that the history issue can be fixed by changing numValues in BGHistory.java (not dividing by 2) here:
https://github.com/NightscoutFoundation/xDrip/blob/436998c61b488f394bff1924511a57b8f4bbeb69/app/src/main/java/com/eveningoutpost/dexdrip/BGHistory.java#L151

**Reply by Navid200** [reply]
Awesome,
I don't have a Libre.  So, even if I make this change, I will not be able to test it.
@jakobheyman Do you have a Libre?  Do you mind testing it?

**Reply by jakobheyman** [reply]
@Navid200 I have a Libre and I have done this change in a fork that displays 1 min data and for me the full data is shown in history (but I have done several other changes as well). I can try to test it more properly (just doing that one change to the original xDrip version) within a day or two.
